File: settlement/room/tests/RoomTests.java
package settlement.room.tests;

import settlement.room.main.ROOMS;
import settlement.room.military.artillery.ROOM_ARTILLERY;
import snake2d.util.misc.ACTION;
import view.interrupter.IDebugPanel;
import view.main.VIEW;
import view.sett.IDebugPanelSett;

public final class RoomTests {

	public RoomTests(ROOMS r) {
		for (ROOM_ARTILLERY a : r.ARTILLERY) {
			IDebugPanelSett.add(new ArtilleryTest(a.eplacer));
		}
		
		IDebugPanel.add("production & trade panel", new ACTION() {
			
			@Override
			public void exe() {
				VIEW.inters().popup.show(new UITradeDebug(), null);
			}
		});
	}
	
}

File: LegacySource/settlement/room/tests/RoomTests.cs
using System;
using settlement.room.main;
using settlement.room.military.artillery;
using snake2d.util.misc;
using view.interrupter;
using view.main;
using view.sett;

public sealed class RoomTests
{
    public RoomTests(ROOMS r)
    {
        foreach (ROOM_ARTILLERY a in r.ARTILLERY)
        {
            IDebugPanelSett.Add(new ArtilleryTest(a.eplacer));
        }

        IDebugPanel.Add("production & trade panel", new ACTION(() =>
        {
            VIEW.Inters().Popup.Show(new UITradeDebug(), null);
        }));
    }
}